### ISSUE TITLE
Implement many masks

### ISSUE THREAD

**Issue by lpatiny** [image-js/ij#15]
ImageJ was developed in java and contains a lot of interesting code. The license is BSD meaning that we can if we cite them get strongly inspired by their code.

We would like to have all the masks that are implement in imageJ based on their code

http://imagej.nih.gov/ij/download/tools/source/ij/process/AutoThresholder.java

**Comment by lpatiny** [comment]
It may be quite complex because they change the name of variable from one filter to another and the code should be cleaned up during traduction to javascript.

**Comment by targos** [comment]
https://github.com/image-js/core/issues/65

**Comment by targos** [comment]
Code on github with color formatting: https://github.com/imagej/imagej1/blob/master/ij/process/AutoThresholder.java

**Comment by targos** [comment]
Please implement the Otsu algorithm first

**Comment by cballesteros** [comment]
It is ok.  We wiil have implemented more commonly used auto thresholding algorithms, among them Otsu and IsoData for our next meeting. We will create a new branch for each method. Of course, the first is Otsu.

Regards,

Cristian

**Comment by lpatiny** [comment]
Dear Cristian,

This is really ok. It is important to have them.

Please have a look at the implemented one because we would like to have 
it with 16 bits images.

Have a nice week-end,

Luc

**Comment by cballesteros** [comment]
Hi Luc, do we have a 16 bits histogram implementation?

**Comment by lpatiny** [comment]
Yes histogram allows to specify a maxSlots to get the number of slots you want

https://github.com/image-js/core/blob/master/src/image/compute/histogram.js

So you can do
```image.getHistogram({maxSlots: 65536});```
to have the 16 bits histogram

The test images are now in
```/image-js/core/test/img```
If you are in this folder and would like to see which image are in which format you may just enter (on a macintosh or linux): 
```file *````
And you will get a list like:
```
ecoli.png:                 PNG image data, 1600 x 1200, 8-bit/color RGB, non-interlaced
format:                    directory
grey16.png:                PNG image data, 454 x 340, 16-bit grayscale, non-interlaced
grey8.png:                 PNG image data, 454 x 340, 8-bit grayscale, non-interlaced
```
Some images are with 16bits depth and the new PNG converter of Michael is now keeping this information.
Please note that you need to checkout the code and run:
```npm update```
```npm run build-test```
and
```npm run watch-browser```
In order to test the code directly in the browser `interactive.html`

You should also try to go to the last version of node (node v4.1.1)

You may contact us anytime if you have problems

**Comment by targos** [comment]
A lot of methods have been implemented, thank you !

**Comment by targos** [comment]
@lpatiny I fixed the percentile method but now two other tests are failing. Can you take a look ?

**Comment by cballesteros** [comment]
Yes, sure!
